pallets/eth-bridge/src/lib.rs

```diff
@@ -556,6 +549,69 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        #[pallet::call_index(7)]
+        #[pallet::weight(<T as Config>::WeightInfo::add_eth_tx_hash())]
+        pub fn submit_latest_ethereum_block(
+            origin: OriginFor<T>,
+            author: Author<T>,
+            latest_seen_block: u32,
+            _signature: <T::AuthorityId as RuntimeAppPublic>::Signature,
+        ) -> DispatchResultWithPostInfo {
+            ensure_none(origin)?;
+            ensure!(Self::active_ethereum_range().is_some(), Error::<T>::VotingEnded);
+            ensure!(
+                author_has_cast_event_vote::<T>(&author.account_id) == false,
+                Error::<T>::EventVoteExists
+            );
+
+            let nominated_range =
+                events_helpers::compute_finalised_block_range_for_latest_ethereum_block(
+                    latest_seen_block,
+                );
+            if let Some(events_partition) =
+                events_helpers::discovered_eth_events_partition_factory(nominated_range, Vec::new())
+                    .first()
+            {
+                let mut votes = EthereumEvents::<T>::get(&events_partition);
+                votes.try_insert(author.account_id).map_err(|_| Error::<T>::EventVotesFull)?;
+
+                EthereumEvents::<T>::insert(&events_partition, votes);
+
+                let mut sorted_votes: Vec<(EthereumEventsPartition, usize)> =
```

Comment:
If am not mistaken, we spoke about re-using existing storage items for the "initial" voting round but would it be simpler to reach consensus on the `initialEthereumBlock`, in its own storage (something that keeps track of who saw what block) until `quorum` number of collators have "seen" an ethereum block. Once they have, we can do the active range setup where `selected_range` starts from the `initialEthereumBlock`:
```
ActiveEthereumRange::<T>::put(ActiveEthRange {
                        range: selected_range,
                        partition: 0,
                        // TODO retrieve values from runtime.
                        event_types_filter: Default::default(),
                    });
```
We can even drain the special storage item we used to reach consensus on the initial voting round. What do you think?

Reply:
I think its better if we use the existing data structures. The additional storage is practically dead code after the initial consensus that we need to keep and never use, regardless of it being drained or not.

Reply:
updated to use an independent storage.